slash/utils/python.py

```diff
@@ -41,11 +41,20 @@ def decorator(new_func):
             setattr(decorator, p, orig)
     return decorator
 
+
+def is_generator(func):
```

Comment:
Why not just use inspect.isgeneratorfunction(get_underlying_func(...)) directly? IIRC it returns the same function if not a generator...

Reply:
You are right... Fixed.